include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -187,6 +187,37 @@ def XTenNN_LoadExternalConstOp: XTenNN_Op<"load_external_const", [
 
 def TosaExtension : NativeOpTrait<"TosaExtension">;
 def ElementwiseUnary : NativeOpTrait<"ElementwiseUnary">;
+class ArrayAttr<int n> : ConfinedAttr<ArrayAttr, [ArrayCount<n>]>;
+class I64DenseArrayAttr<int n> : ConfinedAttr<DenseI64ArrayAttr, [DenseArrayCount<n>]>;
+
+def XtenNN_GroupConv2dOp: XTenNN_Op<"group_conv2d", [Pure, TosaExtension]> {
+  let summary = "Grouped convolution with multiple channel outputs per layer";
+  let description = [{
+    Calculates separates convolutions with multiple channel outputs per layer. The number
+    of groups need to be higher than one.
+
+    This operations expects the same TORCH/ONNX's Conv2d datalayout specification for input and
+    weight [NxCxHxW].
```

Comment:
Did you consider separating the `group` as a dimension to make it easier for matching to the `GroupConv` kernel, or to `MemTileChain`?

Reply:
I thought about it and IMO it made sense to keep that as part of the matching process to avoid adding here unrequired restriction/specifications of third-parties